### ISSUE TITLE
OAuthAccountNotLinked get redirect "/api/auth/auth/login?error=OAuthAccountNotLinked"

### ISSUE THREAD

**Issue by shiqocred** [AntonioErdeljac/next-auth-v5-advanced-guide#30]
when i use a same email in different account i get redirect to `"/api/auth/auth/login?error=OAuthAccountNotLinked"` not which should i get redirect to `"/auth/login?error=OAuthAccountNotLinked"`

anyone know how to solve?

i use `"next-auth": "^5.0.0-beta.13"`

**Comment by Camjo0001** [comment]
did you come across a solution for this?

**Comment by shiqocred** [comment]
No, I haven't found a solution yet, but I wanted to ask maybe someone knows

**Comment by eguenou** [comment]
I had the same issue as you using `"next-auth": "^5.0.0-beta.15"`. I downgrade it to `"next-auth": "^5.0.0-beta.4"` as the one used in this repo to make it work.
I hope this would help.

**Comment by Camjo0001** [comment]
This is great, thanks. 

Did you run into any issues while downgrading?

**Comment by eguenou** [comment]
When I downgrade it, I got an error in auth.config.ts about the Provider type not being the expected one.
I recommend you to upgrade auth.js(next-auth) from that version "5.0.0-beta.4" until a version where your providers are working then stick to it. It is like the library is not regression-proof tested.

**Comment by ksi9302** [comment]
Hi guys,

there's an issue relating to this here with some solutions commented
https://github.com/nextauthjs/next-auth/issues/10273

however the above approach will not pass on the query params, instead this will

 if (isApiAuthRoute) {
    if (
      nextUrl.pathname.startsWith("/api/auth/auth/") &&
      nextUrl.searchParams.get("error")
    ) {
      const url = nextUrl;
      url.pathname = nextUrl.pathname.replace("/api/auth/", "");

      return NextResponse.redirect(url);
    }

    return;
  }

**Comment by Kiezgold** [comment]
middleware.ts :

`if (isApiAuthRoute) {
if (
nextUrl.pathname.startsWith("/api/auth/auth/") &&
nextUrl.searchParams.get("error")
) {
const url = nextUrl;
url.pathname = nextUrl.pathname.replace("/api/auth/", "");

  return NextResponse.redirect(url);
}

return;`
